lib/addons/src/hooks/useDOMRect.ts

```diff
@@ -0,0 +1,61 @@
+import { useState, useEffect, RefObject } from 'react';
+import ResizeObserver from 'resize-observer-polyfill';
+import { useCallbackRef } from './useCallbackRef';
+
+const isStateDirty = (a: UseDOMRectBounds, b: UseDOMRectBounds) => {
+  return !Object.keys(a).every((k: keyof UseDOMRectBounds) => a[k] === b[k]);
+};
+
+export interface UseDOMRectBounds {
+  height: number;
+  width: number;
+  left: number;
+  top: number;
+  bottom: number;
+  right: number;
+  x: number;
+  y: number;
+}
+
+const initialState: UseDOMRectBounds = {
+  height: 0,
+  width: 0,
+  left: 0,
+  top: 0,
+  bottom: 0,
+  right: 0,
+  x: 0,
+  y: 0,
+};
+
+export const useDOMRect = <T extends HTMLElement = HTMLDivElement>(ref: RefObject<T>) => {
+  const { ref: element, fn: attachRef } = useCallbackRef<T>();
```

Comment:
I think you're looking to just use `useRef` ?

Reply:
Been reading up and I do need it for delayed mounts

I can however create the ref in the hook and return it to make outside code cleaner